src/hooks/screen/useCreateChat.ts

```diff
@@ -0,0 +1,103 @@
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import {GROUP_INFO} from '../core/constant';
+
+const useCreateChat = () => {
+  const [loadingCreateChat, setLoadingCreateChat] = React.useState(false);
+  const {localDb} = useLocalDatabaseHook();
+  const {goToChatScreen} = useChatUtilsHook();
+  const createSignChat = async (members: string[], selectedUser, from) => {
+    try {
+      setLoadingCreateChat(true);
+      const initChannel = await SignedMessageRepo.createSignedChat(members);
+      const chatData = await createChannelJson(initChannel, selectedUser);
+
+      const channelList = await ChannelList.fromMessageSignedAPI(chatData);
+      channelList.saveIfLatest(localDb);
+      handleMemberSchema(initChannel);
+      setLoadingCreateChat(false);
+      goToChatScreen(channelList, from);
+    } catch (e) {
+      setLoadingCreateChat(false);
+      console.log({e}, 'error create chat');
+    }
+  };
+
+  const createChannelJson = (response, selectedUser) => {
+    const channelWithMember = {...response.channel, members: response.members};
+    const targetRawJson = {
+      type: 'notification.message_new',
+      cid: response?.channel?.id,
+      channel_id: '',
+      channel_type: 'messaging',
+      channel: channelWithMember,
+      created_at: response?.channel,
+      targetName: selectedUser?.user?.name,
+      targetImage: selectedUser?.user?.image
+    };
+    const chatData = {
+      channel: response?.channel,
+      members: response?.members,
+      appAdditionalData: {
+        rawJson: targetRawJson,
+        message: '',
+        targetName: selectedUser?.user?.name,
+        targetImage: selectedUser?.user?.image
+      }
+    };
+    return chatData;
+  };
+
+  const handleMemberSchema = (response) => {
+    try {
+      response?.members?.forEach(async (member) => {
+        const userMember = UserSchema.fromMemberWebsocketObject(member, response?.channel?.id);
+        await userMember.saveOrUpdateIfExists(localDb);
+
+        const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+          response?.channel?.id,
+          uuid(),
+          member
+        );
+        await memberSchema.save(localDb);
+      });
+    } catch (e) {
+      console.log(e, 'error on memberSchema');
+    }
+  };
+
+  const handleAnonymousMessage = async (selectedUser) => {
+    if (!selectedUser?.allow_anon_dm) {
+      SimpleToast.show('This user does not allow anonymous messages');
+      return;
+    }
+
+    try {
+      const response = await getOrCreateAnonymousChannel(selectedUser?.user_id);
+      const chatData = await createChannelJson(response, selectedUser);
+      const channelList = ChannelList.fromMessageAnonymouslyAPI(chatData);
+      await channelList.saveIfLatest(localDb);
+      handleMemberSchema(response);
+      goToChatScreen(channelList, GROUP_INFO);
+    } catch (e) {
+      SimpleToast.show(e || 'Failed to message this user anonymously');
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the `createSignChat` function, the `handleAnonymousMessage` function is also missing error handling for when `getOrCreateAnonymousChannel(selectedUser?.user_id)` or `ChannelList.fromMessageAnonymouslyAPI(chatData)` fails. Consider adding a toast notification in the catch block to inform the user about the error.

```diff
    } catch (e) {
      SimpleToast.show(e || 'Failed to message this user anonymously');
+      SimpleToast.show('Failed to send anonymous message');
    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->